rust/actyx/ax-aql/src/language/query_impl.rs

```diff
@@ -137,11 +137,10 @@ mod tests {
         let map = query.get_used_event_types().collect::<HashMap<_, _>>();
         assert!(!map.is_empty());
 
-        let start_label = Label::try_from("start").expect("should be a valid label");
-        assert!(map.get(&start_label).is_some());
-
-        let done_label = Label::try_from("done").expect("should be a valid label");
-        assert!(map.get(&done_label).is_none());
+        assert!([("start", true), ("done", false)]
+            .into_iter()
+            .flat_map(|(label, exists)| Label::try_from(label).map(|label| (label, exists)))
+            .all(|(label, exists)| map.contains_key(&label) == exists));
```

Comment:
This is quite noisy and yields no information upon failure. I typically write a little helper to make the test clean and expressive.
```rust
let types = query.get_used_event_types().map(|x| x.0).collect::<BTreeSet<_>>();
assert_eq!(types, t(&["start"]));
```
with a definition of `t` that takes a slice and produces a `BTreeSet<Label>`.